### DISCUSSION TITLE
How do I get the file name to display when I upload a document?

### DISCUSSION THREAD

**Discussion by AleksandraKondracka** [rpldy/react-uploady#230] [Q&A]
Hey,
I know <UploadPreview /> supports showing of images once they have been uploaded however how would I go about displaying just the file name/names that have been uploaded? Thanks in advance! My code is below:

```import { useState } from 'react';
import Uploady from "@rpldy/uploady";
import UploadButton, { asUploadButton } from "@rpldy/upload-button";
import UploadPreview from "@rpldy/upload-preview";
import RoundedBox from '../../nets-commons/components/RoundedBox';
import attachment from '../../nets-commons/icons/attachment.png';
import BasicPopup from '../../nets-commons/components/form/BasicPopup';

const FileUpload = ({ props }) => {
  const [popUpVisible, setPopUpVisible] = useState(false)

  const filterBySize = (file) => {
    //filter out files larger than 5MB
    return file.size < 5242880 ? file.size < 5242880 : setPopUpVisible(true);
  };

  const popupTxt = {
    heading: "This file cannot be uploaded",
    information: "You can only upload files up to a maximum of 5MB and they can only be of the following types: .jpeg, .jpg, .png, .tiff, .bmp, .pdf, .doc, .docx, .xls, .xlsx"
  }

  return (
    <div>
      <RoundedBox className="bg-gray-200">
        <Uploady
          fileFilter={filterBySize}
          accept=".jpeg,.jpg,.png,.tiff,.bmp,.pdf,.doc,.docx,.xls,.xlsx">
          <div className="flex md:items-center items-center space-x-24 mt-2">
            <p className="text-sm text-gray-500">(Max 5 Mb)</p>
            <div className="py-3 px-4 bg-white text-black text-sm flex items-center space-x-2 rounded">
              <img className="ml-1" alt="" src={attachment} height="20px" width="20px" />
              <UploadButton />
            </div>
          </div>
          <div className="mt-2">
            <UploadPreview />
          </div>
        </Uploady>
      </RoundedBox>
      <BasicPopup {...popupTxt}
        onOk={() => setPopUpVisible(false)}
        trigger={popUpVisible} />
    </div>
  )
}

export default FileUpload

**Comment by yoavniran** [comment]
Hi @AleksandraKondracka 
UploadPreview by default shows a video/image preview but you can provide it with a custom preview component (PreviewComponent prop) 

This [sandbox](https://codesandbox.io/s/crop-and-upload-react-uploady-fixed-aspect-simple-g7clx?file=/src/App.js:1347-1363) relies on this ability.

When you pass a custom preview component, one of the props is `name` which is the filename of the file being uploaded. At this point you can display it in whatever way you want. 
The story I mentioned above displays the name of the file so you can see this in action

**Reply by AleksandraKondracka** [reply]
Hi @yoavniran thank you for this! This has helped me to finally display the name of the file. I have 2 questions however:

1. I get an error saying `TypeError: Cannot read property 'id' of undefined` -> Is there any way of fixing this issue?
2. When I upload a file it works fine but then when I go to upload again the file doesn't change and it remains the same - do you have a suggestion on how to go about this so that you can keep re-uploading files. (I assume this could be linked to the id issue?)

Many thanks!

**Reply by yoavniran** [reply]
I will check and let you know

**Reply by yoavniran** [reply]
seems there's a bug in latest version 0.13.3 (probably related to: #232) 
I will investigate and issue a fix soon. 
@AleksandraKondracka  In the meantime you should be able to use 0.13.2

**Reply by yoavniran** [reply]
0.13.4 is available and should fix the problem.
